### PR TITLE
SEED/GBR-04 - OrgMixin endpoint updates for Cycles, Datasets, Labels (depends on SEED/GBR-02)

### PR DESCRIPTION
**Pull request by fablet**
#### Definition of Done:
- [ ] Is there appropriate test coverage? (e.g. ChefSpec, Mocha/Chai, Python, etc.)
- [ ] Does this PR require a Selenium test? (e.g. Browser-specific bugs or complicated UI bugs)
- [ ] Does this PR require a regression test? All fixes require a regression test.
- [ ] Does this add new dependencies? If so, does PIP, npm, bower requirements need to be updated?

### CODE REVIEW

File: seed/filtersets.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+# encoding: utf-8
+"""
+:copyright (c) 2014 - 2016, The Regents of the University of California,
+through Lawrence Berkeley National Laboratory (subject to receipt of any
+required approvals from the U.S. Department of Energy) and contributors.
+All rights reserved.  # NOQA
+:author
+
+# TODO
```
Comment (tallus) [inline seed/filtersets.py]: Add Module level Doc string

File: seed/serializers/labels.py [hunk 2]
```diff
@@ -5,58 +5,97 @@
...
 
-    def __init__(self, *args, **kwargs):
-        """
-        Labels always exist in the context of the organization they are
-        assigned to.  This serializer requires that the `super_organization`
-        for the label be passed into the serializer during initialization so
-        that uniqueness constraints involving the `super_organization` can be
-        validated by the serializer.
+    class Meta:
+        """Model, field, kwargs and validator settings for LabelSerializer.
```
Comment (tallus) [inline seed/serializers/labels.py]: Move this above class Meta/fields to top of class definition

File: seed/serializers/labels.py [hunk 3]
```diff
@@ -5,58 +5,97 @@
...
-        if self.inventory:
-            result = self.inventory.filter(
-                labels=obj,
-            ).values_list('id', flat=True)
 
-            # result = self.inventory.filter(
-            #     labels=obj,
-            # ).exists()
-        return result
+# class LabelSerializer(serializers.ModelSerializer):
```
Comment (tallus) [inline seed/serializers/labels.py]: Remove this

File: seed/views/cycles.py [hunk 4]
```diff
@@ -1,24 +1,180 @@
...
+    """
+    serializer_class = CycleSerializer
+    model = Cycle
+    data_name = 'cycles'
+    filter_class = CycleFilterSet
+    queryset = Cycle.objects.annotate(
+        properties_count=Count('propertyview', distinct=True),
+        taxlots_count=Count('taxlotview', distinct=True)
+    )
+    # ToDo: recommend creating FilterSet/filter_class to filter by dates
```
Comment (tallus) [inline seed/views/cycles.py]: fix/remove

File: seed/views/cycles.py [hunk 5]
```diff
@@ -1,24 +1,180 @@
...
+
+    def perform_create(self, serializer):
+        org_id = self.get_organization(self.request)
+        user = self.request.user
+        serializer.save(organization_id=org_id, user=user)
+
+    # def perform_update(self, serializer):
```
Comment (tallus) [inline seed/views/cycles.py]: resolve todo/remove

File: seed/views/labels.py [hunk 6]
```diff
@@ -1,117 +1,284 @@
...
+        queryset=Property.objects.only('id')
+    )
+    prefetch_taxlots = Prefetch(
+        'taxlot_set',
+        queryset=TaxLot.objects.only('id')
+    )
+    queryset = Label.objects.prefetch_related(
+        prefetch_properties,
+        prefetch_taxlots
+    )
```
Comment (tallus) [inline seed/views/labels.py]: What's the effect of losing the order_by here?

File: seed/views/labels.py [hunk 7]
```diff
@@ -1,117 +1,284 @@
...
+
+
```
Comment (tallus) [inline seed/views/labels.py]: Remove

File: seed/views/labels.py (hunk 6 above)
Comment (fablet) [inline seed/views/labels.py]: adding the 'order_by' method is redundant here since the Labels model already has ordering set to name in the Meta options

**Comment by tallus** [comment]
Is this still valid or was it superseded by 4a?

**Comment by fablet** [comment]
This is superseded by 4a for any work related to cycles.  This branch would need some modification before it is ready for submission upstream

**Pull request closed without merging**